payment-request/updateWith-method-pmi-handling-manual.https.html

```diff
@@ -0,0 +1,138 @@
+<!DOCTYPE html>
+<!-- Copyright © 2017 Mozilla and World Wide Web Consortium, (Massachusetts Institute of Technology, ERCIM, Keio University, Beihang). -->
+<meta charset="utf-8">
+<title>Test for validity of payment method identifiers when calling updateWith() method</title>
+<link rel="help" href="https://www.w3.org/TR/payment-request/#updatewith()-method">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+"use strict";
+setup({ explicit_done: true, explicit_timeout: true });
+const validMethod = Object.freeze({
+  supportedMethods: "https://:@wpt.fyi:443/payment-request",
+});
+
+const validMethods = Object.freeze([validMethod]);
+
+const validAmount = Object.freeze({
+  currency: "USD",
+  value: "1.0",
+});
+
+const validTotal = Object.freeze({
+  label: "Default Total",
+  amount: validAmount,
+});
+
+const validShippingOption = {
+  id: "standard",
+  label: "Shipping option",
+  amount: validAmount,
+  selected: true,
+};
+
+const validDetails = Object.freeze({
+  total: validTotal,
+  shippingOptions: [validShippingOption],
+});
+
+const validModifier = [
+  {
+    supportedMethods: "basic-card",
+    total: validTotal,
+  },
+];
+
+// Avoid false positives, this should always pass
+function smokeTest() {
+  new PaymentRequest(validMethods, validDetails);
+}
+
+function manualTest(button, { invalidMethod }) {
+  button.disabled = true;
+  promise_test(async t => {
+    smokeTest();
+    const request = new PaymentRequest(
+      [{ supportedMethods: "basic-card" }],
+      validDetails,
+      { requestShipping: true }
+    );
+    // We test against a valid and an invalid modifier
+    // expecting the invalid one to throw the RangeError.
+    const shippingChangedPromise = new Promise((resolve, reject) => {
+      request.addEventListener(
+        "shippingaddresschange",
+        ev => {
+          const invalidModifier = Object.assign({}, validModifier, {
+            supportedMethods: invalidMethod,
+          });
+          const invalidDetails = Object.assign({}, validDetails, {
+            modifiers: [validModifier, invalidModifier],
+          });
+          try {
+            ev.updateWith(invalidDetails);
+            resolve(); // This is bad.
+          } catch (err) {
+            reject(err); // this is good.
```

Comment:
I am not sure if this is correct or not. Promise reject/resolve callback might not be executed synchronized, if it is not, updateWith() might not emit any errors and finish its work before the callbacks were called.

Reply:
yeah, you are correct here also.